Fixing tests to reduce database locked error #4651

RADXIshan opened the pull request:
This commit addresses the sqlite3.OperationalError: database is locked issue described in #4651.

Changes made:
- Improved isolation in test_merge.py to ensure tests relying on MergeReports output or SQLite operations do not conflict when run in parallel.
- Added caplog safety checks in test_missing_fields to prevent IndexError when log output is not produced, especially under parallel pytest execution.
- Adjusted string comparison logic in test_missing_fields to more robustly handle formatting mismatches in missing field logging.
- Ensured test functions are compatible with pytest-xdist and do not share database or log state unexpectedly.
-Also some more file changes to fix errors

These fixes help stabilize the CI and reduce flaky test failures caused by concurrent access to SQLite during test runs.

joydeep049 commented on test/test_extractor.py:
```diff
@@ -1,9 +1,3 @@
-""" CVE Binary Tool tests for the extractor function """
-from __future__ import annotations
-
 import shutil
```
We would want the license information here.

RADXIshan commented:
> The output files that were added should also be removed. @RADXIshan

Thank you for the review — I truly appreciate the feedback and will address the issues as soon as possible. I'm still quite new to open source contributions, but I'm very eager to learn and improve. I would be extremely grateful for your patience and guidance as I navigate this process. I may make some mistakes along the way without realising it, so your support means a lot.

RADXIshan commented:
> The output files that were added should also be removed. @RADXIshan

I have added the license comments on the mentioned files and i have deleted the output files , am I supposed to delete test.sbom?

terriko reviewed (changes requested):
I'm not sure what's going on here, but it looks like your PR will remove some code and comments that should not be removed, including a `#nosec` directive that I need for bandit.  As it stands, it's un-mergeable. You may need to rebase or recreate it on a more recent branch synced with `origin/main`, I'm not actually sure what went wrong here.

RADXIshan commented:
> I'm not sure what's going on here, but it looks like your PR will remove some code and comments that should not be removed, including a `#nosec` directive that I need for bandit. As it stands, it's un-mergeable. You may need to rebase or recreate it on a more recent branch synced with `origin/main`, I'm not actually sure what went wrong here.

Ok , i am not sure how that happened i will fork the original repo as new repo and delete the one i have right now , because i dont think it is updated and also has some issues as i can see. Thanks a lot for the review

RADXIshan commented:
> I'm not sure what's going on here, but it looks like your PR will remove some code and comments that should not be removed, including a `#nosec` directive that I need for bandit. As it stands, it's un-mergeable. You may need to rebase or recreate it on a more recent branch synced with `origin/main`, I'm not actually sure what went wrong here.

i have used the update branch currently but i dont think it fixed it fully

joydeep049 commented:
> > The output files that were added should also be removed. @RADXIshan
> 
> I have added the license comments on the mentioned files and i have deleted the output files , am I supposed to delete test.sbom?

I think yes

RADXIshan commented:
> > > The output files that were added should also be removed. @RADXIshan
> > 
> > 
> > I have added the license comments on the mentioned files and i have deleted the output files , am I supposed to delete test.sbom?
> 
> I think yes

I have removed test.sbom

terriko reviewed (changes requested):
Not sure why this removes so many tests out of test_extractor.py, but it shouldn't do that.  Also, I'm not sure the creation of the temp db really *does* anything here since we know that our code is a bit too tightly tied to the current location of the cve.db file.

terriko commented:
Hm, on second thought, in the interest in reducing PR clutter, I'm going to close this as un-mergeable so we can focus our code review efforts on other pull requests that are closer to ready.  Thanks for giving it a shot, though!
